### ISSUE TITLE
Extend `=>` to support pragmas

### ISSUE THREAD

**Issue by dom96** [nim-lang/Nim#11707]
For example:

```nim
myFut.callback = () {.gcsafe.} => closeBot(myFut, i)
```

**Comment by krux02** [comment]
Please use the issue template.

**Comment by ringabout** [comment]
This issue could be closed, `=>` already supports pragmas in Nim v1.4.0

```nim
import sugar

type
  Bot = object
    callback: proc () {.gcsafe.}

proc closeBot(fut: Bot, i: int) =
  discard

var myFut = Bot()

myFut.callback = () {.gcsafe.} => closeBot(myFut, 12)
```

**Comment by ringabout** [comment]
I will make a PR to document this.

ref #14200
